src/foundry/client-esm/canvas/sources/rendered-effect-source.d.mts

```diff
@@ -0,0 +1,325 @@
+  layers: RenderingLayers;
+
+  /**
+   * The color of the source as an RGB vector.
+   */
+  colorRGB: [number, number, number] | null;
+
+  /**
+   * PIXI Geometry generated to draw meshes.
+   */
+  protected _geometry: PIXI.Geometry | null;
+
+  /**
+   * Is the rendered source animated?
+   */
+  get isAnimated(): boolean;
+
+  /**
+   * Has the rendered source at least one active layer?
+   */
+  get hasActiveLayer(): boolean;
+
+  /**
+   * Is this RenderedEffectSource a temporary preview?
+   */
+  get isPreview(): boolean;
+
+  /**
+   * A convenience accessor to the background layer mesh.
+   */
+  get background(): PointSourceMesh;
+
+  /**
+   * A convenience accessor to the coloration layer mesh.
+   */
+  get coloration(): PointSourceMesh;
+
+  /**
+   * A convenience accessor to the illumination layer mesh.
+   */
+  get illumination(): PointSourceMesh;
+
+  _initialize(data: Partial<SourceData>): void;
+
+  /**
+   * Decide whether to render soft edges with a blur.
+   */
+  protected _initializeSoftEdges(): void;
+
+  override _configure(changes: Partial<SourceData>): void;
+
+  /**
+   * Configure which shaders are used for each rendered layer.
+   * @privateRemarks Foundry marks this as private then overrides it in `PointVisionSource`
+   */
+  protected _configureShaders(): {
+    background: AdaptiveLightingShader;
+    coloration: AdaptiveLightingShader;
+    illumination: AdaptiveLightingShader;
+  };
+
+  /**
+   * Specific configuration for a layer.
+   */
+  protected _configureLayer(layer: object, layerId: string): void;
+
+  /**
+   * Create the geometry for the source shape that is used in shaders and compute its bounds for culling purpose.
+   * Triangulate the form and create buffers.
+   * @remarks Marked as abstract
+   */
+  protected _updateGeometry(): void;
+
+  /**
+   * Render the containers used to represent this light source within the LightingLayer
+   */
+  drawMeshes(): {
+    background: PIXI.Mesh;
+    coloration: PIXI.Mesh;
+    illumination: PIXI.Mesh;
+  };
+  /**
+   * Create a Mesh for a certain rendered layer of this source.
+   * @param layerId - The layer key in layers to draw
+   * @returns The drawn mesh for this layer, or null if no mesh is required
+   */
+  protected _drawMesh(layerId: string): PIXI.Mesh | null;
+
+  /**
+   * Update shader uniforms used by every rendered layer.
+   */
+  protected _updateCommonUniforms(shader: AbstractBaseShader): void;
+
+  /**
+   * Update shader uniforms used for the background layer.
+   */
+  protected _updateBackgroundUniforms(): void;
+
+  /**
+   * Update shader uniforms used for the coloration layer.
+   */
+  protected _updateColorationUniforms(): void;
+
+  /**
+   * Update shader uniforms used for the illumination layer.
+   */
+  protected _updateIlluminationUniforms(): void;
+
+  /**
+   * Animate the PointSource, if an animation is enabled and if it currently has rendered containers.
+   * @param dt - Delta time.
+   * @remarks Returns
```

Comment:
Is this source text from Foundry? If so please file under v12 documentation issues.